contract/src/main.rs

```diff
@@ -909,6 +869,12 @@ pub extern "C" fn transfer() {
         runtime::revert(NFTCoreError::InvalidAccount);
     }
 
+    if NFTIdentifierMode::Hash == identifier_mode
+        && utils::should_break_up_owned_token_hashes(source_key)
+    {
+        utils::break_up_owned_token_hashes(source_key)
+    }
+
```

Comment:
move below approved check, then determine `if NFTIdentifierMode::Hash == identifier_mode` once, do both source & target in one block

Reply:
Moved into one collective check